drivers/clk/clk-hifiberry-dacpro.c

```diff
@@ -145,6 +146,8 @@ static int clk_hifiberry_dacpro_prepare(struct clk_hw *hw)
 	default:
 		return -EINVAL;
 	}
+	/* wait for SCLK update to be detected by PCM512x codec */
+	usleep_range(5000, 10000);
```

Comment:
I don't understand the patch comment: `msleep(2)` can in principle be of any length, can it not? And then, if the original delay was for 2ms, then 2ms should be enough, right? Then why not put something like `usleep_range(2000, 4000);`?

Reply:
https://www.kernel.org/doc/html/latest/timers/timers-howto.html
msleep(1~20) may not do what the caller intends, and will often sleep longer (~20 ms actual sleep for any value given in the 1~20ms range). In many cases this is not the desired behavior.

I just experimented with 5 and 10ms, it worked. I tried 2-3 and it didn't.

Reply:
@plbossart hmm, what do you mean "`usleep_range(2000, 3000);` didn't work?" You mean, that it still slept longer? How long? Same as `usleep_range(5000, 10000)`? If it slept longer, that would seem like a bug in the kernel to me. If it slept about as long as the latter - that's possibly ok, but still I'd use "2000, 3000." The reason for the same length of sleep might be your kernel specific timer configuration, tick duration etc. If however someone chooses a "better" configuration, e.g. a shorter timer tick, the "2000, 3000" option might produce a better result. And in any case I think doing what *has* to be done instead of *what works just as well* should be a better option. If both work equally well, I think it's better to use the one that also reflects the real situation, the fact that the hardware requires a 2ms delay.

Reply:
yes, that's what I meant. The msleep(2) actually resulted in a much longer sleep, and experimentally at least 5ms were needed.